vidarr-server/src/test/java/ca/on/oicr/gsi/vidarr/server/VeryBadDataIntegrationTest.java

```diff
@@ -0,0 +1,224 @@
+  private static ServerConfiguration config;
+  private static Main main;
+  private static final HttpClient CLIENT =
+      HttpClient.newBuilder().followRedirects(HttpClient.Redirect.NORMAL).build();
+  @ClassRule public static final TemporaryFolder unloadDirectory = new TemporaryFolder();
+
+  private static ServerConfiguration getTestServerConfig(GenericContainer pg) {
+    ServerConfiguration config = new ServerConfiguration();
+    config.setName("vidarr-test");
+    config.setDbHost(pg.getHost());
+    config.setDbName("vidarr-test");
+    config.setDbPass("vidarr-test");
+    config.setDbUser("vidarr-test");
+    config.setDbPort(pg.getFirstMappedPort());
+    config.setPort(8998);
+    config.setUrl("http://localhost:" + config.getPort());
+    config.setOtherServers(new HashMap<>());
+    config.setInputProvisioners(Collections.singletonMap("raw", new RawInputProvisioner()));
+    config.setWorkflowEngines(new HashMap<>());
+    config.setOutputProvisioners(new HashMap<>());
+    config.setRuntimeProvisioners(new HashMap<>());
+    config.setTargets(new HashMap<>());
+    config.setUnloadDirectory(unloadDirectory.getRoot().getAbsolutePath());
+    return config;
+  }
+
+  @BeforeClass
+  public static void setup() throws SQLException {
+    TimeZone.setDefault(TimeZone.getTimeZone("America/Toronto"));
+    config = getTestServerConfig(pg);
+    main = new Main(config);
+    main.startServer(main);
+    RestAssured.baseURI = config.getUrl();
+    RestAssured.port = config.getPort();
+    RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+    defaultParser = Parser.TEXT;
+    MAPPER.registerModule(new JavaTimeModule());
+  }
+
+  @Before
+  public void cleanAndMigrateDB() {
+    final var simpleConnection = new PGSimpleDataSource();
+    simpleConnection.setServerNames(new String[] {config.getDbHost()});
+    simpleConnection.setPortNumbers(new int[] {config.getDbPort()});
+    simpleConnection.setDatabaseName(config.getDbName());
+    simpleConnection.setUser(config.getDbUser());
+    simpleConnection.setPassword(config.getDbPass());
+    var fw = Flyway.configure().dataSource(simpleConnection);
+    fw.load().clean();
+    fw.locations("classpath:db/migration").load().migrate();
+    // we do this because Flyway on its own isn't finding the test data, and it dies when you
+    // try to give it classpath + filesystem locations in one string. We ignore the "missing"
+    // migrations (run in the migrate() call above).
+    fw.locations("filesystem:src/test/resources/db/migration/")
+        .ignoreMissingMigrations(true)
+        .load()
+        .migrate();
+  }
+
+  public List<String> getNaughtyStringList() {
+    List<String> blns = NaughtyStrings.getStrings();
+    // These indices correspond with strings here
+    // https://github.com/minimaxir/big-list-of-naughty-strings/blob/master/blns.json
+    List<Integer> indices =
+        IntStream.concat(IntStream.range(0, 31), IntStream.range(194, 202))
```

Comment:
I would like to request that, as we are testing an API, we also test 431 through 467